### PR TITLE
Remove dead code in parseArrayDynamic

### PR DESCRIPTION
**Pull request by kkrik-es**
Noticed that the code never executed while running tests for #108911

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java [hunk 1]
```diff
@@ -591,24 +591,20 @@ private static void parseArray(DocumentParserContext context, String lastFieldNa
 
     private static void parseArrayDynamic(DocumentParserContext context, String currentFieldName) throws IOException {
         ensureNotStrict(context, currentFieldName);
-        if (context.dynamic() == ObjectMapper.Dynamic.FALSE) {
```
Comment (kkrik-es) [inline server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: This is the removed part, the rest is just shifted.

**Comment by kkrik-es** [comment]
@elasticsearchmachine update branch

**Comment by kkrik-es** [comment]
@elasticsearchmachine update branch

**Comment by kkrik-es** [comment]
run elasticsearch-ci/bwc-snapshots

**Comment by kkrik-es** [comment]
run elasticsearch-ci/8.13.5 / bwc-snapshots

File: server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java (hunk 1 above)
Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: If I run the `DynamicMappingTests#testArraysDynamicFalse()` test then the if statement does seem to be executed.

Comment (kkrik-es) [reply server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: The test still passes though, I wonder if it has any effect in practice.

Comment (kkrik-es) [reply server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: So what happens is that the function calls #parseDynamicValue that also checks for this, returning immediately.

We can remove it to simplify the logic, but it's fine to keep it too.

Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: I think the test doesn't validate that `skipChildren()` is invoked, which I think is the point of the whole if statement (avoid parsing stuff that we're not going to use). Not sure how, but I think the test should somehow assert that `skipChildren()` is invoked.

Comment (kkrik-es) [reply server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java:594]: This is actually needed, I added more tests.. lemme revert for now.

**Pull request closed without merging**